### PR TITLE
[MT-6765] Add account-memberships list

### PR DESCRIPTION
**Pull request by epmkelly**
## Type

* ### Feature
  Implements a new feature

## Description

adds account-memberships list capability

## Dependencies

https://gitlab.elasticpath.com/commerce-cloud/account-management.svc/-/merge_requests/92

## Issues

MT-6765

* Fixes #

## Notes

n/a

### CODE REVIEW

File: src/types/account-memberships.d.ts [hunk 1]
```diff
@@ -1,64 +1,85 @@
...
-    Create(accountId: string, body: AccountMembershipCreateBody, token?: string): Promise<Resource<AccountMembership>>
+  /**
+   * Get all Account Memberships for an account
+   * @param accountId - The ID for the account
+   * @param token - The Bearer token to grant access to the API
+   */
+  All(
+    accountId: string,
+    token?: string
+  ): Promise<ResourceList<AccountMembership>>
```
Comment (steve-r-west) [inline src/types/account-memberships.d.ts:60]: What happens to our endpoints when we add filtering. How would we manage adding those parameters in a way that is safe and not disrupting to js-sdk users. Maybe we need to add a note here as well.

Comment (steve-r-west) [reply src/types/account-memberships.d.ts:60]: I assume that ResourceList just abstracts over our pagination rules.

Comment (dr-ep) [reply src/types/account-memberships.d.ts:62]: If we support filtering we should implement it as methods Limit, Offset methods, etc. You can get this by inheriting from `QueryableResource`. Take a look at `CurrencyEndpoint` for instance.

**Review by dr-ep — CHANGES_REQUESTED** [review]
Awesome work! I really appreciate the proper JSDocs comments. I had one request for change regarding filtering.

File: src/endpoints/account-memberships.js [hunk 2]
```diff
@@ -10,6 +11,27 @@ class AccountMembershipsEndpoint extends BaseExtend {
     )
   }
 
+  All(accountId, token = null, headers = {}) {
```
Comment (dr-ep) [inline src/endpoints/account-memberships.js:14]: You removed token and headers from definitions but they are still here in javascript.

File: src/endpoints/account-memberships.js [hunk 3]
```diff
@@ -10,6 +11,27 @@ class AccountMembershipsEndpoint extends BaseExtend {
...
+    const { includes, sort, limit, offset, filter } = this
```
Comment (dr-ep) [inline src/endpoints/account-memberships.js:15]: You are using filter and other props here, but I don't see how a user can ever set them? Can you add some tests to show how filtering is supposed to be used.

**Review by dr-ep — CHANGES_REQUESTED** [review]

**Pull request merged**